tests/linearizability/model.go

```diff
@@ -170,11 +176,15 @@ func stepPut(state EtcdState, request EtcdRequest, response EtcdResponse) (bool,
 		state.FailedWrite = &request
 		return true, state
 	}
-	if response.Revision <= state.LastRevision {
+	// Put should always increase revision
+	if response.Revision == state.LastRevision {
 		return false, state
 	}
-	if response.Revision != state.LastRevision+1 && state.FailedWrite == nil {
-		return false, state
+	if state.FailedWrite == nil {
```

Comment:
> for a PUT this would be clearer?

Rewrote this code to be consistent with Delete. Goal is to make `state.FailedWrite == nil` to top of checks because it impacts logic a lot. Put logic is much simpler, but I would want to keep the same code template.

> What the guarantee for a failed write? revision can be +1/+0?

Revision should increase by at least 1, need to fix this